### PR TITLE
[QOL-7667] initial CircleCI config

### PR DESCRIPTION
**Pull request by ThrawnCA**

### CODE REVIEW

**Comment by ThrawnCA** [comment]
CircleCI isn't fully free either, will look into Github Actions.

**Comment by duttonw** [comment]
missing circle ci main config file.

**Comment by ThrawnCA** [comment]
> missing circle ci main config file.

Yes, I took it out because we decided to use Github Actions.

File: bin/travis-build.bash [hunk 1]
```diff
@@ -23,9 +23,10 @@ if [ ! -d ckan ]; then
 fi
 cd ckan
 git checkout $CKAN_BRANCH
-python setup.py develop
 pip install -r requirements.txt
 pip install -r dev-requirements.txt
+pip install -r requirement-setuptools.txt
```
Comment (duttonw) [inline bin/travis-build.bash:28]: should setup tools be run before all others?

Comment (ThrawnCA) [reply bin/travis-build.bash:28]: It generally doesn't matter.

**Comment by duttonw** [comment]
can't you just make it run a shell script instead of having it all spelt out in the github actions yaml file?
see https://github.community/t/how-to-execute-a-script-file-using-github-action/16830/4

**Comment by ThrawnCA** [comment]
> can't you just make it run a shell script instead of having it all spelt out in the github actions yaml file?
> see https://github.community/t/how-to-execute-a-script-file-using-github-action/16830/4

I might do that once I've straightened out the necessary steps. I would have liked to just run the existing scripts intended for TravisCI, but they're not entirely compatible, eg Github Actions doesn't provide vanilla Ubuntu.

**Comment by duttonw** [comment]
can you squish all these commits into one or two commits once its all working again.

**Comment by ThrawnCA** [comment]
> can you squish all these commits into one or two commits once its all working again.

Should be possible.

**Comment by ThrawnCA** [comment]
Obsoleted by 'QOL-7667-github-actions' branch.

**Pull request closed without merging**